### PR TITLE
MailChimp integration query components.

### PR DESCRIPTION
**Pull request by budzanowski**
### Information

This is a part of #17434 PR that represents Query components used in MailChimp integration code.
This is provided for review as a separate PR because the original (spike) PR is too big for the review to be comfortable.

To test these elements in action please refer to #17434.

### CODE REVIEW

File: client/extensions/woocommerce/state/sites/settings/email/queryLists.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId === nextProps.siteId ) {
+			return;
+		}
+		this.request( nextProps );
+	}
+
+	request( props ) {
+		if ( ! props.isRequesting ) {
```
Comment (timmyc) [inline client/extensions/woocommerce/state/sites/settings/email/queryLists.jsx]: Maybe ensure `props.siteId` is defined before requesting the lists just to be safe here.

File: client/extensions/woocommerce/state/sites/settings/email/queryLists.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/**
```
Comment (timmyc) [inline client/extensions/woocommerce/state/sites/settings/email/queryLists.jsx]: 🤔  I know many of the components in the `woocommerce` extension use `.js` as their file extension. Not sure if we should stick to that here or ... ?

File: client/extensions/woocommerce/state/sites/settings/email/querySettings.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId === nextProps.siteId ) {
+			return;
+		}
+		this.request( nextProps );
+	}
+
+	request( props ) {
+		if ( ! props.isRequesting ) {
```
Comment (timmyc) [inline client/extensions/woocommerce/state/sites/settings/email/querySettings.jsx]: Same check for `props.siteId` being set here.

File: client/extensions/woocommerce/state/sites/settings/email/querySyncStatus.jsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+	destroyUpdateTimer = () => {
+		if ( this.updateTimer ) {
+			window.clearInterval( this.updateTimer );
+			this.updateTimer = false;
+		}
+	}
+
+	triggerRequest = ( props ) => {
+		if ( ! props.isRequesting ) {
```
Comment (timmyc) [inline client/extensions/woocommerce/state/sites/settings/email/querySyncStatus.jsx]: should this also be checking to see if `isRequestingResync` is true here? Again one more check for `siteId` never hurts 😄

File: client/extensions/woocommerce/state/sites/settings/email/queryLists.jsx (hunk 2 above)
Comment (budzanowski) [reply client/extensions/woocommerce/state/sites/settings/email/queryLists.jsx]: From slack, I can see that we do not care in the end :)

**Comment by budzanowski** [comment]
Corrections made. Any other thoughts?

**Pull request merged**